code/modules/mob/living/carbon/human/species.dm

```diff
@@ -921,6 +924,38 @@
 		else
 			H.throw_alert("nutrition", /obj/screen/alert/starving)
 
+/datum/species/proc/handle_disgust(mob/living/carbon/human/H, var/metabolism_rate)
```

Comment:
maybe this is too much to ask, but this should really be a status effect

https://github.com/tgstation/tgstation/blob/master/code/datums/status_effects/status_effect.dm